Firstly, we want to import our dictionary that use for predict and train our data 

In [24]:
import numpy as np 
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score

Collect our data fron csv file

In [25]:
#load our data from sonar data.cs v
sonar_data = pd.read_csv('sonar data.csv',header=None)
sonar_data.head()
sonar_data.shape

(208, 61)

We have 208 row and 61 column as training data and test data , below command want to show us about statistics

In [26]:
sonar_data.describe()

0           1           2           3           4           5   \
count  208.000000  208.000000  208.000000  208.000000  208.000000  208.000000   
mean     0.029164    0.038437    0.043832    0.053892    0.075202    0.104570   
std      0.022991    0.032960    0.038428    0.046528    0.055552    0.059105   
min      0.001500    0.000600    0.001500    0.005800    0.006700    0.010200   
25%      0.013350    0.016450    0.018950    0.024375    0.038050    0.067025   
50%      0.022800    0.030800    0.034300    0.044050    0.062500    0.092150   
75%      0.035550    0.047950    0.057950    0.064500    0.100275    0.134125   
max      0.137100    0.233900    0.305900    0.426400    0.401000    0.382300   

               6           7           8           9   ...          50  \
count  208.000000  208.000000  208.000000  208.000000  ...  208.000000   
mean     0.121747    0.134799    0.178003    0.208259  ...    0.016069   
std      0.061788    0.085152    0.118387    0.134416  ...    0.012008   
min      0.003300    0.005500    0.007500    0.011300  ...    0.000000   
25%      0.080900    0.080425    0.097025    0.111275  ...    0.008425   
50%      0.106950    0.112100    0.152250    0.182400  ...    0.013900   
75%      0.154000    0.169600    0.233425    0.268700  ...    0.020825   
max      0.372900    0.459000    0.682800    0.710600  ...    0.100400   

               51          52          53          54          55          56  \
count  208.000000  208.000000  208.000000  208.000000  208.000000  208.000000   
mean     0.013420    0.010709    0.010941    0.009290    0.008222    0.007820   
std      0.009634    0.007060    0.007301    0.007088    0.005736    0.005785   
min      0.000800    0.000500    0.001000    0.000600    0.000400    0.000300   
25%      0.007275    0.005075    0.005375    0.004150    0.004400    0.003700   
50%      0.011400    0.009550    0.009300    0.007500    0.006850    0.005950   
75%      0.016725    0.014900    0.014500    0.012100    0.010575    0.010425   
max      0.070900    0.039000    0.035200    0.044700    0.039400    0.035500   

               57          58          59  
count  208.000000  208.000000  208.000000  
mean     0.007949    0.007941    0.006507  
std      0.006470    0.006181    0.005031  
min      0.000300    0.000100    0.000600  
25%      0.003600    0.003675    0.003100  
50%      0.005800    0.006400    0.005300  
75%      0.010350    0.010325    0.008525  
max      0.044000    0.036400    0.043900  

[8 rows x 60 columns]

M mean mine and R to be Rock . We look from data , we know that mines have 111 and Rocks have 97 .

In [27]:
  sonar_data[60].value_counts()

M    111
R     97
Name: 60, dtype: int64

For the next step, we need to seperate and label our data in two dimensional 

In [28]:
X = sonar_data.drop(columns=60, axis =1)
Y = sonar_data[60]

Train Model to be Logistic Regression

In [29]:
X_train,X_test, Y_train,Y_test = train_test_split(X,Y,test_size=0.45,random_state =0)
print(X_train.shape)

(114, 60)


Next step, we want to make a Logistic Regression Model

In [30]:
Model = LogisticRegression()
Model.fit(X_train,Y_train)

LogisticRegression()

Evaluation Model

In [31]:
#make a prediction from x_train
Y_prediction = Model.predict(X_test)
print("Accuracy will be ",accuracy_score(Y_test,Y_prediction))

Accuracy will be  0.7872340425531915


Make a input and show the prediction

In [32]:
input_data = (0.0307,0.0523,0.0653,0.0521,0.0611,0.0577,0.0665,0.0664,0.1460,0.2792,0.3877,0.4992,0.4981,0.4972,0.5607,0.7339,0.8230,0.9173,0.9975,0.9911,0.8240,0.6498,0.5980,0.4862,0.3150,0.1543,0.0989,0.0284,0.1008,0.2636,0.2694,0.2930,0.2925,0.3998,0.3660,0.3172,0.4609,0.4374,0.1820,0.3376,0.6202,0.4448,0.1863,0.1420,0.0589,0.0576,0.0672,0.0269,0.0245,0.0190,0.0063,0.0321,0.0189,0.0137,0.0277,0.0152,0.0052,0.0121,0.0124,0.0055)
#changing input_data as numpy array
np_array=np.asarray(input_data)
np_reshape = np_array.reshape(1,-1)
prediction = Model.predict(np_reshape)

if prediction[0] == 'M':
  print("It is Mine")
else:
  print("It is Rock")

It is Mine
